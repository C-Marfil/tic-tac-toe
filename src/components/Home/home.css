h3 {
    color: #ffff;
}

.home {
margin-top: 220px;
}
