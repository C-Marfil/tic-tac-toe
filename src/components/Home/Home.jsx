/* eslint-disable no-alert */
import React, { useState } from "react";
import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";
import { Button } from "@chakra-ui/react";
import "../Chat/chat.css";
import "./home.css";

const Home = ({ setUsername, username }) => {
  const navigate = useNavigate();
  const [error, setError] = useState(false);

  const handleUsername = (event) => {
    event.preventDefault();
    setUsername(event.target.value);
  };

  return (
    <div className="home">
      <div className="joinChatContainer">
        <input
          type="text"
          aria-label="username-input"
          placeholder="Enter your name"
          onChange={(event) => {
            handleUsername(event);
          }}
        />
        <Button
          colorScheme="brand"
          type="button"
          onClick={() => {
            if (username) {
              navigate("/lobby");
            } else {
              setError(true);
            }
          }}
        >
          PLAY!
        </Button>
        {error && <p>Please enter your username before proceeding</p>}
      </div>
    </div>
  );
};

Home.propTypes = {
  setUsername: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired,
};

export default Home;
