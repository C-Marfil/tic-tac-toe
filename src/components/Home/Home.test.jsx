import React from "react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Home from "./Home";

describe("Home Component, children and Routing render/work properly", () => {
  const validProps = {
    setUsername: jest.fn(),
    username: "",
  };

  const setup = () => {
    const utils = render(
      <Home
        setUsername={validProps.setUsername}
        username={validProps.username}
      />,
      { wrapper: BrowserRouter }
    );
    const input = screen.getByLabelText("username-input");
    return {
      input,
      ...utils,
    };
  };

  it("Snapshot test", () => {
    const { asFragment } = render(
      <Home
        setUsername={validProps.setUsername}
        username={validProps.username}
      />,
      { wrapper: BrowserRouter }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("tests username input", () => {
    const { input } = setup();
    fireEvent.change(input, { target: { value: "Zufa" } });
    expect(input.value).toBe("Zufa");
  });

  it("tests the component renders", () => {
    render(
      <Home
        setUsername={validProps.setUsername}
        username={validProps.username}
      />,
      { wrapper: BrowserRouter }
    );

    expect(screen.getByText(/PLAY!/i).toBeInTheDocument);
  });

  it("tests button and error message rendering", () => {
    render(
      <Home
        setUsername={validProps.setUsername}
        username={validProps.username}
      />,
      { wrapper: BrowserRouter }
    );
    userEvent.click(screen.getByText(/PLAY!/i));
    expect(
      screen.getByText(/Please enter your username before proceeding/i)
    ).toBeInTheDocument();
  });
});
