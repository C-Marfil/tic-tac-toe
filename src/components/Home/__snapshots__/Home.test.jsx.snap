// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component, children and Routing render/work properly Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="joinChatContainer"
    >
      <input
        aria-label="username-input"
        placeholder="Enter your name"
        type="text"
      />
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        PLAY!
      </button>
    </div>
  </div>
</DocumentFragment>
`;
